### PR TITLE
feat: [playground]Add scrollToPosition functionality to Editor component

### PR DESCRIPTION
**Pull request by smadixd**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?
This PR handles adding functionality to the playground where the user would have the ability to click on a row:col of an alert and would focus and redirection selection/cursor to that position.

#### What changes did you make? (Give an overview)

Changes to the playground code editor component to inject an extra functionality to scroll to a position.

#### Related Issues

#587

### CODE REVIEW

File: src/playground/components/CodeEditor.js [hunk 1]
```diff
@@ -10,33 +10,72 @@ import {
...
+			[eslintOptions, eslintInstance],
+		);
+
+		useImperativeHandle(ref, () => ({
+			scrollToPosition(line, col) {
+				if (editorRef.current) {
+					const editorView = editorRef.current.view;
+					if (editorView) {
+						const { state } = editorView;
+						const pos = state.doc.line(line).from + col;
```
Comment (mdjermanovic) [inline src/playground/components/CodeEditor.js:35]: Column where the cursor is after clicking the 2:11 link seems to be off-by-one:

image

Note that in ESLint messages both lines and columns are 1-based.

Comment (smadixd) [reply src/playground/components/CodeEditor.js:35]: so in this case it should be after the b character?

Comment (mdjermanovic) [reply src/playground/components/CodeEditor.js:35]: The other way, it should be before the left quote.

File: src/playground/components/CodeEditor.js [hunk 2]
```diff
@@ -10,33 +10,72 @@ import {
...
+				bracketMatching(),
+				linter(esLint(eslintInstance, eslintOptions), { delay: 0 }),
+				javascript(),
+				ESLintPlaygroundTheme,
+				ESLintPlaygroundHighlightStyle,
+			],
+			[eslintOptions, eslintInstance],
+		);
+
+		useImperativeHandle(ref, () => ({
```
Comment (amareshsm) [inline src/playground/components/CodeEditor.js:29]: Is it necessary to use ImperativeHandle for this use case? 

Instead, we could store the line and column in a state variable and pass it as a prop to the code editor. In the code editor component, we can use a regular ref to get the editor view and then use the passed props for the line and column to point to the specific line. Are there any drawbacks to this approach?

Comment (smadixd) [reply src/playground/components/CodeEditor.js:29]: Using `useImperativeHandle` allows the parent component to directly control the editor, ensuring precise and immediate scrolling and focusing. Using `state` and `useEffect` can lead to issues with redundant state updates and synchronization, such as when clicking the same row and column twice. This approach keeps the scrolling logic encapsulated in the `CodeEditor`.

Comment (amareshsm) [reply src/playground/components/CodeEditor.js:29]: cool

**Review by mdjermanovic — DISMISSED** [review]
LGTM, thanks! Leaving open for @amareshsm to verify, and would also like @snitin315 to verify.

File: src/playground/components/CodeEditor.js [hunk 3]
```diff
@@ -10,33 +10,72 @@ import {
...
+							const offset =
+								linePos.top -
+								editorRect.top -
+								editorRect.height / 2;
+							editorView.scrollDOM.scrollTop += offset;
+						}
+
+						editorView.focus();
+					}
+				}
```
Comment (amareshsm) [inline src/playground/components/CodeEditor.js:60]:
```suggestion
                const editorView = editorRef.current?.view;
                if (editorView) {
                    const { state } = editorView;
                    const pos = state.doc.line(line).from + col;

                    // Set the cursor selection to the position
                    editorView.dispatch({
                        selection: { anchor: pos },
                        scrollIntoView: true,
                    });

                    const linePos = editorView.coordsAtPos(
                        state.doc.line(line).from,
                    );

                    // Calculate to try to center the line in the editor
                    if (linePos) {
                        const editorRect =
                            editorView.dom.getBoundingClientRect();
                        const offset =
                            linePos.top -
                            editorRect.top -
                            editorRect.height / 2;
                        editorView.scrollDOM.scrollTop += offset;
                    }

                    editorView.focus();
                }
```

Removed Unnecessary Null Checks for editorRef.current and editorRef.current.view

**Review by amareshsm — APPROVED** [review]
LGTM, Thanks. Waiting for Nithin's review.

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**